feat(a11y): enable ECharts aria-label

frassinier opened the pull request:
<!---
Please write the PR title following the conventions at https://www.conventionalcommits.org/en/v1.0.0/
Example:
fix(dashboard): load charts correctly
-->

### SUMMARY

Increasing accessibility support in Apache/Superset through enhanced a11y features, such as generating [aria-label with Echarts](https://echarts.apache.org/handbook/en/best-practices/aria/), is crucial for ensuring inclusivity and usability for all users, including those with disabilities. 

By adhering to [WCAG guidelines](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Perceivable/Use_of_color) and making adjustments to colors and patterns as required, we not only enhance the user experience for individuals with disabilities but also align with standards for accessibility, promoting a more inclusive and equitable platform for all users.

```tsx
        aria: {
          enabled: true,
          decal: {
            show: true,
          },
        },
```

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

<img width="1672" alt="image" src="https://github.com/apache/superset/assets/18534166/304db5ca-488b-4966-b9da-ddc080011743">

### TESTING INSTRUCTIONS
<!--- Required! What steps can be taken to manually verify the changes? -->

### ADDITIONAL INFORMATION
<!--- Check any relevant boxes with "x" -->
<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

frassinier committed:
feat(a11y): enable aria description for echarts

rusackas commented:
I'm not sure we want to turn on the patterns by default. A lot of people don't want/need them, and this provides no option to disable them. We could, of course, add a chart control, but I don't think that's much better. I think the right approach might be to wait until we have [this SIP](https://github.com/apache/superset/issues/28047) in effect, so a user can choose to apply this across the board for charts/dashboards they view/edit/create.

frassinier commented:
@rusackas, I agree! but what do you think of an automated aria label, maybe under a feature flag?

rusackas commented:
Sorry this slid under the radar. I think we're all for aria-label additions, and we can review the idea of the patterns when we have the User Preferences feature (currently in SIP process). This needs a rebase, if you don't mind doing so... hopefully we can get it across the finish line!

frassinier commented:
@rusackas, all set!

rusackas commented:
I think for now, whilst we await the User Preference SIP, we should just add this as a config flag so admins can turn it on and off. Would love to get this merged if we can do so :)

rusackas committed:
fix(a11y): default ECharts aria-label to off to avoid behavior change

Per review discussion, keep this opt-in via theme overrides until the
User Preferences SIP lands, rather than changing default behavior for
all deployments.

rusackas reviewed (approved):
Default-off change looks right, and it's opt-in via theme override until the User Preferences work lands. CI's green (those cypress/npm failures were the same infra flake going around today, cleared on rerun). Thanks for staying with this one, @frassinier.

sadpandajoe committed:
Merge branch 'master' into patch-1
